### PR TITLE
[all_tests][core][dashboard] Add specific server side error messages to server error on 500 response

### LINKED ISSUE #32879: [core][state] Better error message for incorrect node id/ip with ray logs
### What happened + What you expected to happen

If node id or node ip results in a node not found, currently the state api returns 

```

Traceback (most recent call last):
  File "/data/home/rickyx/anaconda3/bin/ray", line 8, in <module>
    sys.exit(main())
  File "/data/home/rickyx/anaconda3/lib/python3.9/site-packages/ray/scripts/scripts.py", line 2422, in main
    return cli()
  File "/data/home/rickyx/anaconda3/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/data/home/rickyx/anaconda3/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/data/home/rickyx/anaconda3/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/data/home/rickyx/anaconda3/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/data/home/rickyx/anaconda3/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/data/home/rickyx/anaconda3/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/data/home/rickyx/anaconda3/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/data/home/rickyx/anaconda3/lib/python3.9/site-packages/ray/experimental/state/state_cli.py", line 930, in log_cluster
    logs = list_logs(
  File "/data/home/rickyx/anaconda3/lib/python3.9/site-packages/ray/experimental/state/api.py", line 1260, in list_logs
    r.raise_for_status()
  File "/data/home/rickyx/anaconda3/lib/python3.9/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http://127.0.0.1:8265/api/v0/logs?node_id=c40ab736a3fcc50ca5e51f6671b3fe8e43030cab0ebe378ab0da165&glob=%2A&timeout=30

```
We should check the validity of the params as early as possible, and hopefully return more specific error info. 

### Versions / Dependencies

master

### Reproduction script

```
pip install ray[default]
ray start --head
ray logs cluster --node-id XXXX 
```

### Issue Severity

None

### PR DESCRIPTION
**Pull request by DACUS1995**
## Why are these changes needed?
Improves the error message for a 500 server error response.

## Related issue number
Closes #32879

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rickyyx** [comment]
Hey @DACUS1995 thanks! 

Can you attach some manual printout for the original issue (wrong node id/ip) to the PR as part of testing? 

@alanwguo @rkooo567  any suggestions on how we could test this or thoughts on this change? 
I am not sure if we have anything that depends on the generic `Server Error` or any case where the generic server error is preferred.

File: dashboard/modules/state/state_head.py
Comment (rickyyx) [inline dashboard/modules/state/state_head.py:392]: nit, what if we use the `error_message` field as reason of `json_reponse` later?

**Review by rickyyx — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/gifs/reactionseditor-yes-awesome-3ohzdIuqJoo8QdKlnW">GIPHY</a></div>

**Comment by rkooo567** [comment]
@rickyyx is there any additional test needed before merging the PR?

**Comment by rickyyx** [comment]
hmm, why it doesn't trigger a full run.

**Comment by rickyyx** [comment]
Also, I think there's a linter error @DACUS1995 
https://buildkite.com/ray-project/oss-ci-build-pr/builds/14211#0186c8eb-9437-4b29-89f2-fa7afd718f64

**Comment by alanwguo** [comment]
> Hey @DACUS1995 thanks!
> 
> Can you attach some manual printout for the original issue (wrong node id/ip) to the PR as part of testing?
> 
> @alanwguo @rkooo567 any suggestions on how we could test this or thoughts on this change? I am not sure if we have anything that depends on the generic `Server Error` or any case where the generic server error is preferred.

The dashboard UI does not do any error checking today. It simply prints errors as they are received from the server.

File: dashboard/modules/state/state_head.py
Comment (DACUS1995) [reply dashboard/modules/state/state_head.py:392]: So basically something like this ?:
```python
    return aiohttp.web.json_response(
        {
            "result": success,
            "msg": message,
            "data": to_google_style(kwargs) if convert_google_style else kwargs,
        },
        dumps=functools.partial(json.dumps, cls=CustomEncoder),
        headers=headers,
        status=200 if success else 500,
        reason=None if success else message,
    )
```
without adding the additional parameter `reason`.

In this case, should behave the same, the question I guess would be: do we always expect the reason to be the same as the error message?

Comment (rickyyx) [reply dashboard/modules/state/state_head.py:392]: > In this case, should behave the same, the question I guess would be: do we always expect the reason to be the same as the error message?

Good question - I don't have a strong opinion on this though. I think as long as it's interepred the same on the client side its fine.

**Comment by rickyyx** [comment]
@DACUS1995 let me know when this is ready for another review!

**Comment by ollie-iterators** [comment]
python/ray/tests/test_state_api_log.py:670:45: E712 comparison to False should be 'if cond is False:' or 'if not cond:'

**Comment by DACUS1995** [comment]
@rickyyx added a small test to check the http exception for invalid node_id under an existing test function `test_log_list`. What are your thoughts on this? Thanks!

**Review by rickyyx — APPROVED** [review]

**Comment by rickyyx** [comment]
Awesome - looks good to me!

cc @rkooo567 for merging.

**Comment by rkooo567** [comment]
Will merge after the feature freeze

**Comment by rickyyx** [comment]
Should be mergeable now cc @rkooo567

**Comment by rkooo567** [comment]
Awesome! Thanks for the contribution!!

**Pull request merged**
